src/multi_vector_simulator/C0_data_processing.py

```diff
@@ -1518,47 +1521,66 @@ def receive_timeseries_from_csv(
         # if only filename is given here, then only one column can be in the csv
         file_name = dict_asset[FILENAME]
         unit = dict_asset[UNIT] + "/" + UNIT_HOUR
-    else:
+    elif FILENAME in dict_asset[input_type]:
         file_name = dict_asset[input_type][FILENAME]
         header = dict_asset[input_type][HEADER]
         unit = dict_asset[input_type][UNIT]
+    else:
+        load_from_timeseries_instead_of_file = True
 
     file_path = os.path.join(settings[PATH_INPUT_FOLDER], TIME_SERIES, file_name)
-    C1.lookup_file(file_path, dict_asset[LABEL])
-
-    data_set = pd.read_csv(file_path, sep=",", keep_default_na=True)
+    if os.path.exists(file_path) is False:
+        msg = (
+            f"Missing file! The timeseries file '{file_path}' \nof asset "
+            + f"{dict_asset[LABEL]} can not be found."
+        )
+        logging.warning(msg + " Looking for {TIMESERIES} entry.")
+        # if the file is not found
+        load_from_timeseries_instead_of_file = True
 
-    if FILENAME in dict_asset:
-        header = data_set.columns[0]
+    else:
+        data_set = pd.read_csv(file_path, sep=",", keep_default_na=True)
+
+    # If loading the data from the file does not work (file not present), the data might be
+    # already present in dict_values under TIMESERIES
+    if load_from_timeseries_instead_of_file is False:
+        if FILENAME in dict_asset:
+            header = data_set.columns[0]
+        series_values = data_set[header]
+    else:
+        if TIMESERIES in dict_asset:
+            series_values = dict_asset[TIMESERIES]
+        else:
+            FileNotFoundError(msg)
```

Comment:
The `msg` is redundant and will be printed twice, almost identically. Maybe add suffix  like `"TIMESERIES data is not provided."`

Reply:
It will not be printed twice as if the file exists, then `load_from_timeseries_instead_of_file` value is `False`